source/slang/slang-ir-dce.cpp

```diff
@@ -120,6 +120,15 @@ struct DeadCodeEliminationContext
             UInt operandCount = inst->getOperandCount();
             for( UInt ii = 0; ii < operandCount; ++ii )
             {
+                switch (inst->getOp())
+                {
+                case kIROp_BoundInterfaceType:
+                    if (inst->getOperand(ii)->getOp() == kIROp_WitnessTable)
+                        continue;
+                    break;
+                default:
+                    break;
+                }
```

Comment:
It would be good to eliminate special cases like this from the DCE pass, by making sure the source instructions actually represent the dependencies of the code. Allowing things to be removed as dead code even when they are referenced like this seems like it could break some assumptions/invariants in the DCE pass.